src/appkit.cs

```diff
@@ -4705,27 +4832,82 @@ partial interface NSControl {
 		void EndEditing ([NullAllowed] NSText textObj);
 	}
 
-	[DesignatedDefaultCtor]
-	[BaseType (typeof (NSObject))]
-	interface NSController : NSCoding {
+	[Protocol]
+	interface NSEditorRegistration
+	{
 		[Export ("objectDidBeginEditing:")]
-		void ObjectDidBeginEditing (NSObject editor);
+		void ObjectDidBeginEditing (INSEditor editor);
 
 		[Export ("objectDidEndEditing:")]
-		void ObjectDidEndEditing (NSObject editor);
+		void ObjectDidEndEditing (INSEditor editor);
+	}
+
+	[Category]
+	[BaseType (typeof(NSObject))]
+	interface NSObject_NSEditorRegistration
+	{
+		[Export ("objectDidBeginEditing:")]
+		void ObjectDidBeginEditing (INSEditor editor);
+
+		[Export ("objectDidEndEditing:")]
+		void ObjectDidEndEditing (INSEditor editor);
+	}
+	
+	interface INSEditor {}
 
+	[Protocol]
+	interface NSEditor
```

Comment:
This part of the diff is tricky, parts of NSController were pulled up into NSEditor, but not all / exactly.